packages/grafana-data/src/field/fieldDisplay.test.ts

```diff
@@ -156,4 +157,64 @@ describe('FieldDisplay', () => {
     const display = getFieldDisplayValues(options);
     expect(display[0].field.thresholds!.length).toEqual(1);
   });
+
+  it('Should return field mapped value when there is no data', () => {
+    const mapEmptyToText = '0';
+
+    const options: GetFieldDisplayValuesOptions = {
+      data: [
+        {
+          name: 'No data',
+          fields: [],
+          length: 0,
+        },
+      ],
+      replaceVariables: (value: string) => {
+        return value;
+      },
+      fieldOptions: {
+        calcs: [],
+        override: {
+          mappings: [
+            {
+              id: 1,
+              operator: '',
+              text: mapEmptyToText,
+              type: MappingType.ValueToText,
+              value: 'null',
+            },
+          ],
+        },
+        defaults: {},
+      },
+      theme: {} as GrafanaTheme,
+    };
```

Comment:
Nitpick: maybe move this to a separate `testContext|testSetup|setup` function and reuse some of the code in both test cases?

Reply:
Yeah, I was thinking about this as well. But I have a couple of other ideas that I would like to check with you guys. But suuper good feedback!

I am a big fan of the builder pattern when it comes to creating test-data. Please let me know if this is something we should talk more about.

Reply:
Yes, please share